backend/src/main/scala/sbt/internal/inc/bloop/internal/BloopNameHashing.scala

```diff
@@ -84,7 +84,10 @@ private final class BloopNameHashing(
           initialChangedSources,
           allSources.map(v => v: VirtualFileRef),
           previous
-        ).collect { case f: VirtualFile => f }
```

Comment:
`initialChagedSources` are received as `VirtualFileRef` and `mapInvalidationToSources` doesn't lift them into `VirtualFile`.
Using `collect` here is incorrect as it just drops correctly invalidated sources.

Reply:
och! That looks like the issue i've been having! Awesom!